### PR TITLE
[AzureMonitorExporter] fix log exception bug

### PR DESCRIPTION
**Pull request by TimothyMothra**
Currently logging an exception will override the exception message.
This change fixes that bug.
This has a side effect of changing how AggregateExceptions are logged.
I verified that these changes are more in line with Application Insights SDK.

## Changes
- changed `LogHelper` to consider `Exception.Message`
- reenable test `E2ETelemetryItemValidation/LogsTests.cs` `VerifyException`
- updated all affected tests

## Examples of AI SDK

<details>
  <summary>Example of ILogger + Exception </summary>
  
```csharp
[TestMethod]
public void Test_LogException()
{
    // SETUP
    var channel = new TestChannel();

    var loggerFactory = LoggerFactory.Create(builder =>
    {
        builder.AddApplicationInsights(
            configureTelemetryConfiguration: config =>
            {
                config.InstrumentationKey = "testIkey";
                config.TelemetryChannel = channel;
            },
            options => { }
        );
    });
    var logger = loggerFactory.CreateLogger("TestLogCategory");

    // ACT
    try
    {
        throw new Exception("Test Exception");
    }
    catch (Exception ex)
    {
        logger.Log(
            logLevel: LogLevel.Error,
            eventId: 0,
            exception: ex,
            message: "Hello {name}.",
            args: new object[] { "World" });
    }

    // ASSERT
    Assert.IsTrue(channel.TelemetryItems.Any());

    var telemetryItem = channel.TelemetryItems.First() as ExceptionTelemetry;
    Assert.AreEqual("Test Exception", telemetryItem.Message);
    Assert.AreEqual("Hello World.", telemetryItem.Properties["FormattedMessage"]);
}
```

image

</details>

<details>
    <summary>Example of Aggregate Exception</summary>

```csharp
        [TestMethod]
        public void Test_LogAggregateException()
        {
            // SETUP
            var channel = new TestChannel();

            var loggerFactory = LoggerFactory.Create(builder =>
            {
                builder.AddApplicationInsights(
                    configureTelemetryConfiguration: config =>
                    {
                        config.InstrumentationKey = "testIkey";
                        config.TelemetryChannel = channel;
                    },
                    options => { }
                );
            });
            var logger = loggerFactory.CreateLogger("TestLogCategory");

            // ACT
            int numberOfExceptions = 15;
            int maxNumberOfExceptionsAllowed = 10;
            List<Exception> innerExceptions = new List<Exception>();
            for (int i = 0; i < numberOfExceptions; i++)
            {
                innerExceptions.Add(new Exception((i + 1).ToString(CultureInfo.InvariantCulture)));
            }

            AggregateException rootLevelException = new AggregateException("0", innerExceptions);

            // Passing "0" explicitly here instead of using rootLevelException.Message
            // rootLevelException.Message will return different value in case of net461 compared to netcore
            logger.LogWarning(rootLevelException, "0");

            // ASSERT
            Assert.IsTrue(channel.TelemetryItems.Any());
            var telemetryItem = channel.TelemetryItems.First() as ExceptionTelemetry;

            var first_ExceptionDetails = telemetryItem.Exceptions[0];
            Assert.AreEqual("System.AggregateException", first_ExceptionDetails.typeName);
            Assert.AreEqual("0 (1) (2) (3) (4) (5) (6) (7) (8) (9) (10) (11) (12) (13) (14) (15)", first_ExceptionDetails.message);


        }
```

image

</details>

### CODE REVIEW

**Comment by TimothyMothra** [comment]
Net461 is failing. need more time to investigate

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs [hunk 1]
```diff
@@ -297,10 +297,16 @@ public void AggregateExceptionsWithMultipleNestedExceptionsAreTrimmedAfterReachi
 
             Assert.Equal(maxNumberOfExceptionsAllowed + 1, exceptionData.Exceptions.Count);
 
-            for (int counter = 0; counter < maxNumberOfExceptionsAllowed; counter++)
+            // The first exception is the AggregateException
+            Assert.Equal("System.AggregateException", exceptionData.Exceptions[0].TypeName);
+            Assert.Equal("0 (1) (2) (3) (4) (5) (6) (7) (8) (9) (10) (11) (12) (13) (14) (15)", exceptionData.Exceptions[0].Message);
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs:302]: This behavior is different in net461 versus netcore. probably need conditional compile?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs:320]: Done.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs [hunk 2]
```diff
@@ -19,6 +19,55 @@ namespace Azure.Monitor.OpenTelemetry.Exporter.Tests
...
+        [InlineData(false)]
+        public void MessageIsSetToExceptionMessage(bool parseStateValues)
+        {
+            var logRecords = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.IncludeFormattedMessage = true;
+                    options.ParseStateValues = parseStateValues;
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs:33]: FYI - this is by default to `true` now. https://github.com/Azure/azure-sdk-for-net/pull/34433

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs [hunk 3]
```diff
@@ -232,7 +232,13 @@ public void ExceptionDataContainsExceptionDetailsofAllInnerExceptionsOfAggregate
             var exceptionData = new TelemetryExceptionData(2, logRecords[0]);
 
             Assert.Equal(3, exceptionData.Exceptions.Count);
+
+#if NET461
             Assert.Equal("AggregateException", exceptionData.Exceptions[0].Message);
+#else
+            Assert.Equal("AggregateException (Inner1) (Inner2)", exceptionData.Exceptions[0].Message);
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs:239]: Could we jut assert on `aggregateException.Message`?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs (hunk 1 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs:320]: Same as https://github.com/Azure/azure-sdk-for-net/pull/33860/files#r1136312001

**Review by vishweshbankwar — APPROVED** [review]
LGTM with minor suggestion

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs (hunk 3 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs:239]: Yes! that's much clearer. Thanks!!

**Comment by vishweshbankwar** [comment]
@TimothyMothra - Could you please also update the changelog?

**Pull request merged**